Multiple files: reduce the number of include statements

mykaul opened the pull request:
We have hundreds of redundant, or unused #include statements in the code. I have a WIP patch to remove many of them
We have several that are circular (include A calls B which calls A ...)
We have some unused statements in popular include file (glusterd.h is a good example).
The end goal would be to reduce the no. of include statements, include what's needed,
and thus improve code readability, maintainability and hopefully reduce compilation times.

Note: it is still not pretty, but it's a good step towards really including what is needed, per file/modulex/xlator/...

Updates: #3130
Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

Commit:
Multiple files: reduce the number of include statements

We have hundreds of redundant, or unused #include statements in the code. I have a WIP patch to remove many of them
We have several that are circular (include A calls B which calls A ...)
We have some unused statements in popular include file (glusterd.h is a good example).
The end goal would be to reduce the no. of include statements, include what's needed,
and thus improve code readability, maintainability and hopefully reduce compilation times.

Note: it is still not pretty, but it's a good step towards really including what is needed, per file/modulex/xlator/...

Updates: #3130
Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/api/src/glfs-primary.c b/api/src/glfs-primary.c
index 1d6474e16..21257164b 100644
--- a/api/src/glfs-primary.c
+++ b/api/src/glfs-primary.c
@@ -10,7 +10,6 @@
 
 #include <stdio.h>
 
-
 #include "glfs-internal.h"
 #include "glfs-mem-types.h"
 #include "gfapi-messages.h"
diff --git a/libglusterfs/src/common-utils.c b/libglusterfs/src/common-utils.c
index 407e9814f..374553a2e 100644
--- a/libglusterfs/src/common-utils.c
+++ b/libglusterfs/src/common-utils.c
@@ -5458,7 +5458,7 @@ gf_pipe(int fd[2], int flags)
     int ret = 0;
 #if defined(HAVE_PIPE2)
     ret = pipe2(fd, flags);
-#else /* not HAVE_PIPE2 */
+#else  /* not HAVE_PIPE2 */
     ret = pipe(fd);
     if (ret < 0)
         return ret;
diff --git a/libglusterfs/src/gf-io-common.c b/libglusterfs/src/gf-io-common.c
index b12bc4790..2120c22e3 100644
--- a/libglusterfs/src/gf-io-common.c
+++ b/libglusterfs/src/gf-io-common.c
@@ -498,8 +498,8 @@ gf_io_thread_join(pthread_t thread, struct timespec *timeout)
 {
 #ifdef GF_LINUX_HOST_OS
     if (timeout != NULL) {
-        gf_io_success(gf_io_call_ret(pthread_timedjoin_np, thread, NULL,
-                                     timeout));
+        gf_io_success(
+            gf_io_call_ret(pthread_timedjoin_np, thread, NULL, timeout));
 
         return;
     }
diff --git a/libglusterfs/src/glusterfs/default-args.h b/libglusterfs/src/glusterfs/default-args.h
index 384d21693..341d0c762 100644
--- a/libglusterfs/src/glusterfs/default-args.h
+++ b/libglusterfs/src/glusterfs/default-args.h
@@ -15,7 +15,6 @@
 #ifndef _DEFAULT_ARGS_H
 #define _DEFAULT_ARGS_H
 
-
 int
 args_lookup_cbk_store(default_args_cbk_t *args, int32_t op_ret,
                       int32_t op_errno, inode_t *inode, struct iatt *buf,
diff --git a/xlators/cluster/afr/src/afr-dir-read.c b/xlators/cluster/afr/src/afr-dir-read.c
index d63aa694e..1510a7e3a 100644
--- a/xlators/cluster/afr/src/afr-dir-read.c
+++ b/xlators/cluster/afr/src/afr-dir-read.c
@@ -228,9 +228,8 @@ afr_readdir_cbk(call_frame_t *frame, void *cookie, xlator_t *this,
     }
 
     if (op_ret >= 0)
-        op_ret = afr_readdir_transform_entries(frame, subvol_entries,
-                                               (long)cookie, &entries,
-                                               local->fd);
+        op_ret = afr_readdir_transform_entries(
+            frame, subvol_entries, (long)cookie, &entries, local->fd);
 
     AFR_STACK_UNWIND(readdir, frame, op_ret, op_errno, &entries, xdata);
 
diff --git a/xlators/cluster/afr/src/afr-inode-read.c b/xlators/cluster/afr/src/afr-inode-read.c
index 883dc2d0f..2c7576397 100644
--- a/xlators/cluster/afr/src/afr-inode-read.c
+++ b/xlators/cluster/afr/src/afr-inode-read.c
@@ -17,7 +17,7 @@
 
 #include <urcu/uatomic.h>
 
-#include "libxlator.h" // for gf_get_max_stime()
+#include "libxlator.h"  // for gf_get_max_stime()
 #include <glusterfs/dict.h>
 #include <glusterfs/logging.h>
 #include <glusterfs/list.h>
diff --git a/xlators/cluster/afr/src/afr-lk-common.c b/xlators/cluster/afr/src/afr-lk-common.c
index 75f6e14f6..f1e8b8042 100644
--- a/xlators/cluster/afr/src/afr-lk-common.c
+++ b/xlators/cluster/afr/src/afr-lk-common.c
@@ -8,7 +8,6 @@
   cases as published by the Free Software Foundation.
 */
 
-
 #include "afr-transaction.h"
 #include "afr-messages.h"
 
diff --git a/xlators/cluster/ec/src/ec-fops.h b/xlators/cluster/ec/src/ec-fops.h
index 8e50467f3..c9ce3afe4 100644
--- a/xlators/cluster/ec/src/ec-fops.h
+++ b/xlators/cluster/ec/src/ec-fops.h
@@ -11,7 +11,6 @@
 #ifndef __EC_FOPS_H__
 #define __EC_FOPS_H__
 
-
 #include "ec-types.h"
 #include "ec-common.h"
 
diff --git a/xlators/features/changelog/lib/src/gf-changelog-rpc.h b/xlators/features/changelog/lib/src/gf-changelog-rpc.h
index 176fd7361..3c5269f62 100644
--- a/xlators/features/changelog/lib/src/gf-changelog-rpc.h
+++ b/xlators/features/changelog/lib/src/gf-changelog-rpc.h
@@ -11,7 +11,6 @@
 #ifndef __GF_CHANGELOG_RPC_H
 #define __GF_CHANGELOG_RPC_H
 
-
 #include "gf-changelog-helpers.h"
 #include "changelog-rpc-common.h"
 
diff --git a/xlators/features/changelog/src/changelog-encoders.h b/xlators/features/changelog/src/changelog-encoders.h
index 9cbc2f942..272396a37 100644
--- a/xlators/features/changelog/src/changelog-encoders.h
+++ b/xlators/features/changelog/src/changelog-encoders.h
@@ -11,7 +11,6 @@
 #ifndef _CHANGELOG_ENCODERS_H
 #define _CHANGELOG_ENCODERS_H
 
-
 #include "changelog-helpers.h"
 
 #define CHANGELOG_STORE_ASCII(priv, buf, off, gfid, gfid_len, cld)             \
diff --git a/xlators/features/changelog/src/changelog-misc.h b/xlators/features/changelog/src/changelog-misc.h
index 38969f1ab..30b1fd6b4 100644
--- a/xlators/features/changelog/src/changelog-misc.h
+++ b/xlators/features/changelog/src/changelog-misc.h
@@ -11,7 +11,6 @@
 #ifndef _CHANGELOG_MISC_H
 #define _CHANGELOG_MISC_H
 
-
 #define CHANGELOG_MAX_TYPE 4
 #define CHANGELOG_FILE_NAME "CHANGELOG"
 #define HTIME_FILE_NAME "HTIME"
diff --git a/xlators/features/selinux/src/selinux.h b/xlators/features/selinux/src/selinux.h
index 4db649d45..58d59ac11 100644
--- a/xlators/features/selinux/src/selinux.h
+++ b/xlators/features/selinux/src/selinux.h
@@ -10,7 +10,6 @@
 #ifndef __SELINUX_H__
 #define __SELINUX_H__
 
-
 #define SELINUX_XATTR "security.selinux"
 #define SELINUX_GLUSTER_XATTR "trusted.glusterfs.selinux"
 
diff --git a/xlators/mgmt/glusterd/src/glusterd-op-sm.c b/xlators/mgmt/glusterd/src/glusterd-op-sm.c
index efc40e599..d64b6fad5 100644
--- a/xlators/mgmt/glusterd/src/glusterd-op-sm.c
+++ b/xlators/mgmt/glusterd/src/glusterd-op-sm.c
@@ -125,8 +125,8 @@ glusterd_txn_opinfo_dict_fini()
 
 void
 glusterd_txn_opinfo_init(glusterd_op_info_t *opinfo,
-                         glusterd_op_sm_state_t state, int *op,
-                         dict_t *op_ctx, rpcsvc_request_t *req)
+                         glusterd_op_sm_state_t state, int *op, dict_t *op_ctx,
+                         rpcsvc_request_t *req)
 {
     glusterd_conf_t *conf = NULL;
 
@@ -5820,9 +5820,9 @@ glusterd_op_sm_transition_state(glusterd_op_info_t *opinfo,
     conf = THIS->private;
     GF_ASSERT(conf);
 
-    (void)glusterd_sm_tr_log_transition_add(
-        &conf->op_sm_log, opinfo->state, state[event_type].next_state,
-        event_type);
+    (void)glusterd_sm_tr_log_transition_add(&conf->op_sm_log, opinfo->state,
+                                            state[event_type].next_state,
+                                            event_type);
 
     opinfo->state = state[event_type].next_state;
     return 0;
diff --git a/xlators/mgmt/glusterd/src/snapshot/glusterd-lvm-snapshot.c b/xlators/mgmt/glusterd/src/snapshot/glusterd-lvm-snapshot.c
index 6b8b49af8..ead7ff6a5 100644
--- a/xlators/mgmt/glusterd/src/snapshot/glusterd-lvm-snapshot.c
+++ b/xlators/mgmt/glusterd/src/snapshot/glusterd-lvm-snapshot.c
@@ -556,7 +556,7 @@ glusterd_lvm_brick_details(dict_t *rsp_dict,
         if (!token) {
             ret = -1;
             gf_msg(this->name, GF_LOG_ERROR, EINVAL, GD_MSG_INVALID_ENTRY,
-                    "Invalid vg entry");
+                   "Invalid vg entry");
             goto end;
         }
         value = gf_strdup(token);
@@ -572,7 +572,7 @@ glusterd_lvm_brick_details(dict_t *rsp_dict,
         ret = dict_set_dynstr(rsp_dict, key, value);
         if (ret) {
             gf_msg(this->name, GF_LOG_ERROR, 0, GD_MSG_DICT_SET_FAILED,
-                    "Could not save vgname ");
+                   "Could not save vgname ");
             goto end;
         }
 
diff --git a/xlators/mgmt/glusterd/src/snapshot/glusterd-zfs-snapshot.c b/xlators/mgmt/glusterd/src/snapshot/glusterd-zfs-snapshot.c
index b8699c632..86e51bdfc 100644
--- a/xlators/mgmt/glusterd/src/snapshot/glusterd-zfs-snapshot.c
+++ b/xlators/mgmt/glusterd/src/snapshot/glusterd-zfs-snapshot.c
@@ -486,9 +486,9 @@ glusterd_zfs_snap_clone_brick_path(char *snap_mount_dir,
                        snap_brick_dir);
     else {
         if (restore)
-            len = snprintf(brickinfo->path, sizeof(brickinfo->path), "%s/%s/brick%d%s",
-                           snap_mount_dir, snap_clone_volume_id, brick_num,
-                           snap_brick_dir);
+            len = snprintf(brickinfo->path, sizeof(brickinfo->path),
+                           "%s/%s/brick%d%s", snap_mount_dir,
+                           snap_clone_volume_id, brick_num, snap_brick_dir);
         else
             len = snprintf(brickinfo->path, sizeof(brickinfo->path),
                            "%s/.zfs/snapshot/%s_%d/%s", origin_brick_mount,
diff --git a/xlators/system/posix-acl/src/posix-acl.c b/xlators/system/posix-acl/src/posix-acl.c
index 0a9ae4758..ded706406 100644
--- a/xlators/system/posix-acl/src/posix-acl.c
+++ b/xlators/system/posix-acl/src/posix-acl.c
@@ -8,7 +8,6 @@
   cases as published by the Free Software Foundation.
 */
 
-
 #include <glusterfs/defaults.h>
 
 #include "posix-acl.h"
```

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/xlators/cluster/afr/src/afr-lk-common.c b/xlators/cluster/afr/src/afr-lk-common.c
index 75f6e14f6..f1e8b8042 100644
--- a/xlators/cluster/afr/src/afr-lk-common.c
+++ b/xlators/cluster/afr/src/afr-lk-common.c
@@ -8,7 +8,6 @@
   cases as published by the Free Software Foundation.
 */
 
-
 #include "afr-transaction.h"
 #include "afr-messages.h"
 
diff --git a/xlators/features/changelog/src/changelog-encoders.h b/xlators/features/changelog/src/changelog-encoders.h
index 9cbc2f942..272396a37 100644
--- a/xlators/features/changelog/src/changelog-encoders.h
+++ b/xlators/features/changelog/src/changelog-encoders.h
@@ -11,7 +11,6 @@
 #ifndef _CHANGELOG_ENCODERS_H
 #define _CHANGELOG_ENCODERS_H
 
-
 #include "changelog-helpers.h"
 
 #define CHANGELOG_STORE_ASCII(priv, buf, off, gfid, gfid_len, cld)             \
```

mykaul commented:
/run regression

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/libglusterfs/src/common-utils.c b/libglusterfs/src/common-utils.c
index 7a1e80069..35e5b3ddf 100644
--- a/libglusterfs/src/common-utils.c
+++ b/libglusterfs/src/common-utils.c
@@ -293,7 +293,8 @@ gf_gfid_generate_from_xxh64(uuid_t gfid, char *key)
     }
 
     gf_msg_debug(this->name, 0,
-                 "gfid generated is %s (hash1: %" XXH64_FMT ") "
+                 "gfid generated is %s (hash1: %" XXH64_FMT
+                 ") "
                  "hash2: %" XXH64_FMT ", xxh64_1: %s xxh64_2: %s",
                  uuid_utoa(gfid), hash_1, hash_2, xxh64_1, xxh64_2);
 
diff --git a/xlators/protocol/client/src/client-common.h b/xlators/protocol/client/src/client-common.h
index d1a251fa3..620939a3b 100644
--- a/xlators/protocol/client/src/client-common.h
+++ b/xlators/protocol/client/src/client-common.h
@@ -15,7 +15,6 @@
 #include "glusterfs3.h"
 #include "client.h"
 
-
 /* New functions for version 4 */
 int
 client_post_common_dict(xlator_t *this, gfx_common_dict_rsp *rsp, dict_t **dict,
```

mykaul commented:
/run regression

mykaul commented:
/run regression

Commit:
Multiple files: reduce the number of include statements (cont.)

We have hundreds of redundant, or unused #include statements in the code. I have a WIP patch to remove many of them
We have several that are circular (include A calls B which calls A ...)
We have some unused statements in popular include file (glusterd.h is a good example).
The end goal would be to reduce the no. of include statements, include what's needed,
and thus improve code readability, maintainability and hopefully reduce compilation times.

Note: it is still not pretty, but it's a good step towards really including what is needed, per file/modulex/xlator/...

Updates: #3130
Signed-off-by: Yaniv Kaul <ykaul@redhat.com>

mykaul commented:
/run regression

mykaul commented:
/run regression

mykaul commented:
/recheck smoke

amarts commented:
340 files? I just checked that all of them are straight forward. LGTM. Whats your take @pranithk @xhernandez. If no objection in next 2 days, will merge this.

mykaul commented:
> 340 files? I just checked that all of them are straight forward. LGTM. Whats your take @pranithk @xhernandez. If no objection in next 2 days, will merge this.

It started small, but no one reviewed it, so I kept removing.
¯\_(ツ)_/¯

mohit84 commented:
> > 340 files? I just checked that all of them are straight forward. LGTM. Whats your take @pranithk @xhernandez. If no objection in next 2 days, will merge this.
> 
> It started small, but no one reviewed it, so I kept removing. ¯_(ツ)_/¯

I think all changes related to include statement and changes should verifiy during compile time. If the code is compiled successfully we can assume things are ok.
